lua/lvim/lsp/manager.lua

```diff
@@ -62,47 +60,52 @@ local function client_is_configured(server_name, ft)
   return false
 end
 
+local function launch_server(server_name, config)
+  pcall(function()
+    require("lspconfig")[server_name].setup(config)
+    buf_try_add(server_name)
+  end)
+end
+
 ---Setup a language server by providing a name
 ---@param server_name string name of the language server
----@param user_config table [optional] when available it will take predence over any default configurations
+---@param user_config table? when available it will take predence over any default configurations
 function M.setup(server_name, user_config)
   vim.validate { name = { server_name, "string" } }
+  user_config = user_config or {}
 
   if lvim_lsp_utils.is_client_active(server_name) or client_is_configured(server_name) then
     return
   end
 
-  local config = resolve_config(server_name, user_config)
-
   local servers = require "nvim-lsp-installer.servers"
-  local server_available, requested_server = servers.get_server(server_name)
+  local server_available, server = servers.get_server(server_name)
 
   if not server_available then
-    pcall(function()
-      require("lspconfig")[server_name].setup(config)
-      buf_try_add(server_name)
-    end)
+    local config = resolve_config(server_name, user_config)
+    launch_server(server_name, config)
     return
   end
 
-  local install_notification = false
+  local install_in_progress = false
 
-  if not requested_server:is_installed() then
+  if not server:is_installed() then
     if lvim.lsp.automatic_servers_installation then
       Log:debug "Automatic server installation detected"
-      requested_server:install()
-      install_notification = true
+      server:install()
+      install_in_progress = true
     else
-      Log:debug(requested_server.name .. " is not managed by the automatic installer")
+      Log:debug(server.name .. " is not managed by the automatic installer")
     end
   end
 
-  requested_server:on_ready(function()
-    if install_notification then
-      vim.notify(string.format("Installation complete for [%s] server", requested_server.name), vim.log.levels.INFO)
+  server:on_ready(function()
+    if install_in_progress then
+      vim.notify(string.format("Installation complete for [%s] server", server.name), vim.log.levels.INFO)
     end
-    install_notification = false
-    requested_server:setup(config)
+    install_in_progress = false
+    local config = resolve_config(server_name, server:get_default_options(), user_config)
```

Comment:
I don't believe you need to retrieve nvim-lsp-installer's options here. This will all be applied automatically by nvim-lsp-installer behind the scenes, or am I missing something?

```diff
-    local config = resolve_config(server_name, server:get_default_options(), user_config)
+    local config = resolve_config(server_name, user_config)
```

Reply:
@williamboman, it's the same problem as https://github.com/LunarVim/LunarVim/issues/2507, I'm worried that the setup is called too late, since `lvim.lsp.manager` is usually called from an ftplugin file.

Instead I want to do the setup more imperatively, to avoid relying on load order.